### PR TITLE
Rename internal columns breaking change

### PR DESCRIPTION
**Pull request by SferaDev**
- Rename `id` to `xata_id` breaking change as well as other internal columns.
- Update types and tests.
- Remove `xata` object and `getMetadata`
- Update search columns

### CODE REVIEW

File: packages/client/src/schema/record.ts [hunk 1]
```diff
@@ -191,15 +158,15 @@ type EditableDataFields<T> = T extends XataRecord
   ? number | NumericOperator
   : T;
 
-export type EditableData<O extends XataRecord> = Identifiable &
-  Partial<
+export type EditableData<O extends XataRecord> = Partial<
+  Identifiable &
```
Comment (SferaDev) [inline packages/client/src/schema/record.ts:162]:
```suggestion
export type EditableData<O extends XataRecord> = Identifiable &
  Partial<
```

File: packages/client/src/schema/record.ts [hunk 2]
```diff
@@ -191,15 +158,15 @@ type EditableDataFields<T> = T extends XataRecord
...
     Omit<
       {
         [K in keyof O]: EditableDataFields<O[K]>;
       },
       keyof XataRecord
     >
-  >;
+>;
```
Comment (SferaDev) [inline packages/client/src/schema/record.ts:169]:
```suggestion
  >;
```

File: packages/client/src/schema/repository.ts [hunk 3]
```diff
@@ -1977,13 +1982,13 @@ export class RestRepository<Record extends XataRecord>
 
     for (const [key, value] of Object.entries(object)) {
       // Ignore internal properties
-      if (key === 'xata') continue;
+      if (['xata_version', 'xata_createdat', 'xata_updatedat'].includes(key)) continue;
```
Comment (SferaDev) [inline packages/client/src/schema/repository.ts:1986]:
```suggestion
```

**Review by eemmiillyy — APPROVED** [review]
LGTM

File: packages/client/src/schema/selection.ts [hunk 4]
```diff
@@ -163,7 +151,7 @@ type NestedColumns<O, RecursivePath extends any[]> = RecursivePath['length'] ext
     >;
 
 // Private: Utility type to get object properties without XataRecord ones
-type DataProps<O> = Exclude<StringKeys<O>, StringKeys<XataRecord>>;
+type DataProps<O> = Exclude<StringKeys<O>, StringKeys<Omit<XataRecord, 'xata_id'>>>;
```
[the remark this answers is not in the document]
Comment (SferaDev) [reply packages/client/src/schema/selection.ts:154]: Nope, since `xata_id` is the only one we add to `XataRecord` via `Identifiable`

**Pull request merged**